File: Algorithms/Graph_Algorithms/BFS/BFS.py
def bfs(graph, start):
    visited = [False] * len(graph)
    queue = []
    print(start)
    visited[start] = True
    queue.insert(0, start)
    while(len(queue) != 0):
        actual = queue.pop()
        for i in range(0, len(graph)):
            if(graph[actual][i] > 0 and not visited[i]):
                print(i)
                visited[i] = True
                queue.insert(0, i)

"""
    Graph is represented with an Adjacency Matrix.

    Negative values mean that there's no edge between
    the two nodes, and 0 means it's the same node.

    Positive values mean that there's an edge between
    the two nodes.
"""
graph = [ [0,1,1,-1], [1,0,1,-1], [1,1,0,1], [-1,-1,1,0] ]
bfs(graph, 0)

File: Algorithms/Graph_Algorithms/BFS/BFS.java
import java.util.LinkedList;
import java.util.Queue;

public class Graph {
    
    int[][] graph;
    
    public Graph(int[][] g) {
        graph = g;
    }
    
    public void bfs(int start) {
        int actual;
        boolean[] visited = new boolean[graph.length];
        Queue<Integer> q = new LinkedList<Integer>();
        System.out.println(start);
        visited[start] = true;
        q.offer(start);
        while(!q.isEmpty()) {
            actual = q.poll();
            for(int i = 0; i < graph.length; i++) {
                if(graph[actual][i] > 0 && !visited[i]) {
                    System.out.println(i);
                    visited[i] = true;
                    q.offer(i);
                }
            }
        }
    }
    
    public static void main(String[] args) {
        Graph g = new Graph(new int[][]{
                                        {0,1,1,-1},
                                        {1,0,1,-1},
                                        {1,1,0,1},
                                        {-1,-1,1,0} });
        g.bfs(0);
    }
    
}
